src/test/denialofservice_tests.cpp

```diff
@@ -109,9 +109,17 @@ BOOST_AUTO_TEST_CASE(outbound_slow_chain_eviction)
         dummyNode1.vSendMsg.clear();
     }
 
-    int64_t nStartTime = GetTime();
-    // Wait 21 minutes
-    SetMockTime(nStartTime+21*60);
+    int64_t time{GetTime()};
+    // Wait 10 minutes
+    time += 10 * 60;
+    SetMockTime(time);
+
+    // Clear nPingNonceSent so we don't disconnect for a ping timeout
+    dummyNode1.nPingNonceSent = 0;
```

Comment:
Move the test change to its own commit so it's easy to verify behaviour doesn't change with the move logic to net_processing commit? I don't follow why the test change is necessary/desirable though.

Reply:
good idea. Will do.

The change is required because previously ping timeouts happened in in CConnman, so it was fine to bump the time in this test because there's no CConnman to time out the connection. When the ping data and timout logic is moved to PeerManager, then the connection times out in this test if you bump the time forwards by 21 minutes without resetting the ping nonce.

Reply:
Makes sense; might be a useful commit message for the test change?

Seems kind-of annoying to have to remember to constrain your mocktime bumps by random ping timeout limits. Shouldn't this also (already) cause problems in functional tests?

Reply:
The functional tests have an asyncio event loop thread that will automatically respond to ping messages:

https://github.com/bitcoin/bitcoin/blob/f656165e9c0d09e654efabd56e6581638e35c26c/test/functional/test_framework/p2p.py#L392-L393

The unit tests are single threaded, so any logic to handle messages needs to be done in the main test logic.

Reply:
ok, split this into its own commit.